### ISSUE TITLE
DOM doesn't re-render when navigating to sibling dynamic route

### ISSUE THREAD

**Issue by stordahl** [sveltejs/sapper#1766]
**Describe the bug**
I have a Sapper app which has a side bar linking to each dynamic route (/poems/[slug]). When the user is on a route like /poems/poem-1 and clicks an anchor tag linked to /poems/poem-2, the route will change but the DOM does not update. However, if the user refreshes the page the correct content loads on refresh.

**Logs**
Unfortunately, there are no server or browser console errors

**To Reproduce**
To see this bug, we have a branch deploy [here](https://test--karlakratz.netlify.app/poems/psyche)
You can also view the source code [here](https://github.com/karlabrowne/karlakratzpoetry)

**Expected behavior**

**Stacktraces**
If you have a stack trace to include, we recommend putting inside a `<details>` block for the sake of the thread's readability:

<details>
  <summary>Stack trace</summary>

  Stack trace goes here...
</details>

**Information about your Sapper Installation:**

  System:
    OS: macOS 10.15.7
    CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
    Memory: 405.93 MB / 8.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 15.11.0 - /usr/local/bin/node
    npm: 7.6.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 90.0.4430.85
    Firefox Developer Edition: 88.0
    Safari: 13.1.3
  npmPackages:
    rollup: ^2.3.4 => 2.45.2 
    sapper: ^0.29.1 => 0.29.1 
    svelte: ^3.17.3 => 3.37.0 

Browser: Firefox Developer Edition: 88.0

Hosted on Netlify running `npm run export`. Locally, I've also ran a build and saw the same resulting bug.

**Severity**
Basically makes Sapper unusable; I'm surprised this hasn't been raised in the past?
